programs/sol-xr/src/instructions/invest.rs

```diff
@@ -0,0 +1,102 @@
+use {
+    crate::state::sol_strategy::SolStrategy,
+    anchor_lang::prelude::Rent,
+    anchor_lang::prelude::*,
+    anchor_lang::system_program,
+    anchor_spl::{
+        associated_token::AssociatedToken,
+        token::{mint_to, Mint, MintTo, Token, TokenAccount},
+    },
+};
+
+#[derive(Accounts)]
+pub struct Invest<'info> {
+    #[account(mut)]
+    pub investor: Signer<'info>,
+
+    #[account(
+        mut,
+        seeds = [b"mint"],
+        bump
+    )]
+    pub mint: Account<'info, Mint>,
+
+    #[account(
+        init_if_needed,
+        payer = investor,
+        associated_token::mint = mint,
+        associated_token::authority = investor,
+    )]
+    pub associated_token_account: Account<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        seeds = [SolStrategy::SEED_PREFIX],
+        bump
+    )]
+    pub sol_strategy: Account<'info, SolStrategy>,
+
+    pub token_program: Program<'info, Token>,
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    pub system_program: Program<'info, System>,
+}
+
+pub fn invest_handler(ctx: Context<Invest>, amount: u64) -> Result<()> {
+    // Return error if amount would lead to an address going over the initial pool cap
+    let account_size = SolStrategy::INIT_SPACE + 8;
+    let rent_exempt_amount = Rent::get()?.minimum_balance(account_size);
+    let current_strategy_balance =
+        ctx.accounts.sol_strategy.to_account_info().lamports() - rent_exempt_amount;
```

Comment:
there should be a custom state to track deposit to the strategy account, using `lamports()` could lead to inconsistency.